CHANGELOG.md

```diff
@@ -8,9 +8,39 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 A note on future revisions.
 Everything within a major version number should be code compatible (with the exception of experimental interfaces). The most notable example of an experimental interface is the support for multiple source inputs. The APIs to deal with this will change in future minor releases. Everything within a single minor release should be network compatible with other federates on the same minor release number. Compatibility across minor release numbers may be possible in some situations but we are not going to guarantee this as those components are subject to performance improvements and may need to be modified at some point. Patch releases will be limited to bug fixes and other improvements not impacting the public API or network compatibility. Check the [Public API](./docs/Public_API.md) for details on what is included and excluded from the public API and version stability.
 
+## [3.1.0][] - 2021-11-25
+
+This release includes some bug fixes and enhancements to HELICS 3 which is now the default. The [migrating 2 to 3](./docs/developer-guide/porting-2-to-3.md) page includes some specific details on migrating from HELICS 2 to 3.
+
+### Fixed
+
+- Fixed an issue with null strings lacking a null termination when retrieved from the C API. This primarily affected the Python and other language API's.
+- CMake fix for some older linkers.
+- A Fix for ASIO aligned when used with MINGW.
+- Fix to IPC core to support boost 1.77.
+- A few issues around the JsonSerialization options for backwards compatibility with HELICS 2, and add some interoperability tests that can be used to check future versions.
+- Fix and issue with the timeCoordinator where is a large time value was requested and with certain other configuration parameters the time could overflow and cause a deadlock.
+
+### Changed
+
+- precommit-ci now used in the CI chain and checks were updated and a check for python formatting inside the docs.
+- Update thirdParty library versions including ASIO, CLI11, toml11, FMT, spdlog, jsoncpp, filesystem.
+- Mac binaries are now universal binaries to support ARM based CPU types.
+- Update some of the TCP helper classes to better support future encrypted communication efforts
+
+### Added
+
+- Add the ability to add observer federates which can join a federation after it has started and subscribe to values, or make queries of the federation while in progress.
+- Add a configurable grant timeout which will trigger diagnostic action if a federate has been waiting sufficiently long for a time grant
+- a document on the [Type conversion](./docs\developer-guide/typeConversion.md) that HELICS can perform and a series of tests to verify the correct operation of the type conversions
+- Additional missing functions related to linking targeted endpoints from a core, so it can work similar to dataLink methods. These methods were added to Core and Broker applications
```

Comment:
```suggestion
- Additional missing functions related to linking targeted endpoints from a core, so it can work similar to dataLink methods. These methods were added to Core and Broker applications.
```